src/puppetlabs/jdbc_util/pglogical.clj

```diff
@@ -123,11 +123,12 @@
   or :none if replication-mode is :none. Valid status values vary depending on
   thew role; see other replication-status functions for more information."
   [db replication-mode]
-  (case replication-mode
-    :source (provider-replication-status db)
-    :replica (replica-replication-status db)
-    :none :none
-    nil :none))
+  (with-timeout 4 :unknown
```

Comment:
I'd prefer to see the value declared somewhere and used, rather than an embedded magic number.  The other location is problematic as well, although I don't think needs to be addressed here.  Do we have any evidence that 4 seconds is the right value for this?  The docstring should be updated to indicate that a result of :timeout is possible.